File: app/src/main/java/com/cna/mineru/cna/Utils/LoadingDialog.java
package com.cna.mineru.cna.Utils;

import android.animation.ValueAnimator;
import android.app.Activity;
import android.graphics.drawable.AnimationDrawable;
import android.graphics.drawable.ColorDrawable;
import android.os.CountDownTimer;
import android.support.v7.app.AppCompatDialog;
import android.text.TextUtils;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import com.airbnb.lottie.LottieAnimationView;
import com.cna.mineru.cna.R;

public class LoadingDialog {
    AppCompatDialog progressDialog;

    public void progressON(Activity activity, String message) {
        if (activity == null || activity.isFinishing()) {
            return;
        }
        if (progressDialog != null && progressDialog.isShowing()) {
            progressSET(message);
        } else {
            progressDialog = new AppCompatDialog(activity);
            progressDialog.setCancelable(false);
            progressDialog.getWindow().setBackgroundDrawable(new ColorDrawable(android.graphics.Color.TRANSPARENT));
            progressDialog.setContentView(R.layout.progress_page);
            progressDialog.show();
        }

        final LottieAnimationView animationView = (LottieAnimationView) progressDialog.findViewById(R.id.animation_view);
        final ValueAnimator animator = ValueAnimator.ofFloat(0f, 1f).setDuration(1600);
        animator.addUpdateListener(new ValueAnimator.AnimatorUpdateListener() {
            @Override
            public void onAnimationUpdate(ValueAnimator animation) {
                animationView.setProgress((Float) animation.getAnimatedValue());
            }
        });

        TextView tv_progress_message = (TextView) progressDialog.findViewById(R.id.tv_progress_message);
        if (!TextUtils.isEmpty(message)) {
            tv_progress_message.setText(message);
        }

        CountDownTimer cTimer = new CountDownTimer(10000, 1000) {
            public void onTick(long millisUntilFinished) {

            }
            public void onFinish() {
                progressOFF();
            }
        };
        cTimer.start();
    }

    private void progressSET(String message) {
        if (progressDialog == null || !progressDialog.isShowing()) {
            return;
        }

        TextView tv_progress_message = (TextView) progressDialog.findViewById(R.id.tv_progress_message);
        if (!TextUtils.isEmpty(message)) {
            tv_progress_message.setText(message);
        }
    }

    public void progressOFF() {
        if (progressDialog != null && progressDialog.isShowing()) {
            progressDialog.dismiss();
        }
    }
}

File: app/src/main/java/com/cna/mineru/cna/Utils/LoadingDialog.kt
package com.cna.mineru.cna.Utils

import android.animation.ValueAnimator
import android.app.Activity
import android.graphics.drawable.ColorDrawable
import android.os.CountDownTimer
import android.text.TextUtils
import android.widget.TextView
import androidx.appcompat.app.AppCompatDialog

import com.airbnb.lottie.LottieAnimationView
import com.cna.mineru.cna.R

class LoadingDialog {
    internal var progressDialog: AppCompatDialog? = null

    fun progressON(activity: Activity?, message: String) {
        if (activity == null || activity.isFinishing) {
            return
        }
        if (progressDialog != null && progressDialog!!.isShowing()) {
            progressSET(message)
        } else {
            progressDialog = AppCompatDialog(activity)
            progressDialog!!.setCancelable(false)
            progressDialog!!.window?.setBackgroundDrawable(ColorDrawable(android.graphics.Color.TRANSPARENT))
            progressDialog!!.setContentView(R.layout.progress_page)
            progressDialog!!.show()
        }

        val animationView =
            progressDialog!!.findViewById<LottieAnimationView>(R.id.animation_view)
        val animator = ValueAnimator.ofFloat(0f, 1f).setDuration(1600)
        animator.addUpdateListener { animation ->
            animationView?.progress = animation.animatedValue as Float
        }

        val tv_progress_message =
            progressDialog!!.findViewById<TextView>(R.id.tv_progress_message)
        if (!TextUtils.isEmpty(message)) {
            tv_progress_message?.text = message
        }

        val cTimer = object : CountDownTimer(10000, 1000) {
            override fun onTick(millisUntilFinished: Long) {

            }

            override fun onFinish() {
                progressOFF()
            }
        }
        cTimer.start()
    }

    private fun progressSET(message: String) {
        if (progressDialog == null || !progressDialog!!.isShowing()) {
            return
        }

        val tv_progress_message =
            progressDialog!!.findViewById<TextView>(R.id.tv_progress_message)
        if (!TextUtils.isEmpty(message)) {
            tv_progress_message?.text = message
        }
    }

    fun progressOFF() {
        if (progressDialog != null && progressDialog!!.isShowing()) {
            progressDialog!!.dismiss()
        }
    }
}
